src/display/Coordinates/Cartesian.tsx

```diff
@@ -20,13 +18,14 @@ export function Cartesian({
   yAxis: yAxisOverrides,
   subdivisions = false,
 }: CartesianCoordinatesProps) {
+  const idNumber = useId("cartesian")
   const xAxisEnabled = xAxisOverrides !== false
   const yAxisEnabled = yAxisOverrides !== false
 
   const xAxis = { subdivisions, ...defaultAxisOptions, ...xAxisOverrides } as GridAxisOptions
   const yAxis = { subdivisions, ...defaultAxisOptions, ...yAxisOverrides } as GridAxisOptions
 
-  const id = `cartesian-${incrementer++}`
+  const id = `cartesian-${idNumber}`
```

Comment:
Oh, I just realized that Vector was using `useMemo` and this one wasn't. Cartesian was the one giving us trouble; I wonder if a better solution would be to wrap this in `useMemo` like Vector.

Reply:
I don't think a `useMemo` would fix your tests, but I think it'd solve the re-render issue without introducing a new context yeah.